### PR TITLE
Enable typeguard pytest plugin

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context

We have static type guarding, now we can use pytest to do runtime typeguarding too (only when pytest is run). 
It also caught a few minor typing bugs (with lists/ndarrays) used interchangably that have now been fixed.

This will only check if any existing typehints are used incorrectly. (Ex passing a list when it expects a numpy array etc).

## How Has This Been Tested
Typeguard and pytest

## Types of changes

- [X] Docs change / refactoring / dependency upgrade

## Checklist

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] All new and existing tests passed.

### CODE REVIEW

File: habitat_sim/sensors/noise_models/no_noise_model.py [hunk 1]
```diff
@@ -21,7 +22,8 @@
 
 
 @registry.register_noise_model(name="None")
-class NoSensorNoiseModel(SensorNoiseModel):
+@attr.s(auto_attribs=True)
+class NoSensorNoiseModel(SensorNoiseModel):  # type: ignore
```
Comment (Skylion007) [inline habitat_sim/sensors/noise_models/no_noise_model.py:26]: It kept complaining about the fact that the constructor required a GPU_ID due to the parent class, but I couldn't override it. 'attr.s' is deprecated anyhow, we really should be just using Python dataclasses instead.

File: habitat_sim/agent/controls/object_controls.py [hunk 2]
```diff
@@ -16,9 +17,10 @@
 
 # epislon used to deal with machine precision
 EPS = 1e-5
+_3d_point = Union[np.ndarray, mn.Vector3, Tuple[float, float, float]]
 
 
-def _noop_filter(start: np.array, end: np.array) -> np.array:
+def _noop_filter(start: _3d_point, end: _3d_point) -> _3d_point:
```
Comment (eundersander) [inline habitat_sim/agent/controls/object_controls.py:23]: It seems like this noop filter was written to work with any type and is very general-purpose. Is there a good reason to narrow the types so specifically?

Comment (eundersander) [reply habitat_sim/agent/controls/object_controls.py:23]: Er, I guess you're making the type broader, not narrower here. I guess what I'm saying is, both the old typing (nd.array) and the new typing (_3d_point) seem unnecessarily narrow.

**Review by eundersander — APPROVED** [review]
I posted a philosophical question about Python typing, but regardless, LGTM.

**Pull request merged**